### DISCUSSION TITLE
Using Captum with Heterogeneous graph?

### DISCUSSION THREAD

**Discussion by AdrianBZG** [pyg-team/pytorch_geometric#4685] [Q&A]
Hi, is there any way or "hack" to use Captum for interpretability in heterogeneous graphs? There are some resources for homogeneous but I'm unsure on how to approach this for the hetero case.

**Comment by rusty1s** [comment]
The only way I can think of is to make use of `RGCNConv` or `RGATConv` (which operate on a "typed" homogeneous graph with `edge_type` vector). The formulation of heterogeneous graphs as a set of bipartite graphs currently does not support explainability. I am sorry.

**Reply by AdrianBZG** [reply]
@rusty1s Thanks. How can different node types with a RGATConv approach? I currently have 2 node types in my HeteroData, one with features and labels (this one is the set of nodes to predict) and the other with no features and no labels.

**Reply by rusty1s** [reply]
You need to create some features for your second node type (random initialization or mean aggregation of neighbors usually works fine). Then, you can just concatenate the node features of the two types together.

**Reply by AdrianBZG** [reply]
@rusty1s I see, do you have an example for this? I'm not quite sure how the concatenation should be done. Thanks.

**Reply by rusty1s** [reply]
If you have a heterogeneous data object `HeteroData` with two node types:
```python
data['user'].x = ...  # [num_nodes, 128]
data['paper'].x = None
```
you can do
```python
data['paper'].x = torch.randn(num_paper_nodes, 128)
```
and call `to_homogeneous()` afterwards:
```python
data = data.to_homogeneous()
```
This will do the concatenation internally.
